extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/VaultBootstrapConfig.java

```diff
@@ -210,6 +210,17 @@
     @ConfigItem(defaultValue = DEFAULT_READ_TIMEOUT)
     public Duration readTimeout;
 
+    /**
+     * List of remote hosts that are not proxied when the client is configured to use a proxy. This
+     * list serves the same purpose as the JVM {@code nonProxyHosts} configuration.
+     *
+     * <p>
+     * Entries can use the <i>*</i> wildcard character for pattern matching, e.g <i>*.example.com</i> matches
+     * <i>www.example.com</i>.
+     */
+    @ConfigItem
+    public Optional<List<String>> nonProxyHosts;
```

Comment:
if this was just a `List`, and the property was not defined, would that end up being an empty list? if so do we need an optional around the list?

Reply:
You cannot have undefined properties with Lists, you need the Optional.